flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/result/MaterializedCollectBatchResult.java

```diff
@@ -76,10 +77,9 @@ public boolean isMaterialized() {
 	}
 
 	@Override
-	public void startRetrieval(ProgramDeployer deployer) {
-		deployer
-				.deploy()
-				.thenCompose(jobClient -> jobClient.getJobExecutionResult(classLoader))
+	public void startRetrieval(JobClient jobClient) {
+		CompletableFuture.completedFuture(jobClient)
+				.thenCompose(client -> client.getJobExecutionResult(classLoader))
```

Comment:
Should change the exception message since we only retrieve results here.

Reply:
this should just be jobClient.getJobExecutionResult(classLoader), no need to wrap the jobClient in a future and then compose sth to it.